src/core/services/compiler/compiler.js

```diff
@@ -6,7 +6,20 @@
  */
 angular
   .module('material.core')
-  .service('$mdCompiler', MdCompilerService);
+  .service('$mdCompiler', MdCompilerService)
+  .provider('$$mdPreAssignBindings', PreAssignBindingsProvider);
+
+/**
+ * Provider that is used to report the preAssignBindingsEnabled state to the $mdCompiler service.
+ */
+function PreAssignBindingsProvider($compileProvider) {
+  var fallbackValue = angular.version.major === 1 && angular.version.minor < 6;
+  var hasPreAssignState = !!$compileProvider.preAssignBindingsEnabled;
+
+  this.$get = function() {
+    return hasPreAssignState ? $compileProvider.preAssignBindingsEnabled() : fallbackValue;
```

Comment:
This will not catch changes that will happend at runtime (which is theoretically possible, although highly improbable).